cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -514,7 +601,18 @@ func (scaleSet *ScaleSet) DeleteNodes(nodes []*apiv1.Node) error {
 		ref := &azureRef{
 			Name: node.Spec.ProviderID,
 		}
-		refs = append(refs, ref)
+
+		if node.Annotations[cloudprovider.FakeNodeReasonAnnotation] == cloudprovider.FakeNodeUnregistered {
+			klog.V(5).Infof("Node: %s type is unregistered..Appending to the unregistered list", node.Name)
+			unregisteredRefs = append(unregisteredRefs, ref)
```

Comment:
Delete unregistered nodes separately from registered nodes, and with `hasUnregisteredNodes` fixed to `true`?

Reply:
> with hasUnregisteredNodes fixed to true

should make no differences as it should be true whenever there is unregistered nodes in the first place (`hasUnregisteredNodes` has the same trigger condition as the population of `unregisteredRefs`)

Reply:
> Delete unregistered nodes separately

Should not have any issue